tests/main/snap-discard-ns/mount-py2.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python2
+"""
+Pure-python limited-use wrapper around the mount library function.
+"""
+from __future__ import print_function, absolute_import, unicode_literals
+
+from argparse import ArgumentParser
+from ctypes import CDLL, c_char_p, c_long, get_errno
+from ctypes.util import find_library
+from os import strerror
+from sys import stderr
+
+__all__ = ('mount', )
+
+
+class MountOpts(object):
+    """MountOpts contain various flags for the mount system call."""
+
+    Bind = 4096
+
+
+def mount(source, target, fstype, flags=0, data=""):
+    # type: (unicode, unicode, unicode, int, unicode) -> None
+    """mount is a thin wrapper around the mount library function."""
+    libc_name = find_library(b"c")
+    if libc_name is None:
+        raise Exception("cannot find the C library")
+    libc = CDLL(libc_name, use_errno=True)
+    retval = libc.mount(
+            c_char_p(source.encode("UTF-8")),
+            c_char_p(target.encode("UTF-8")),
+            c_char_p(fstype.encode("UTF-8")),
+            c_long(flags),
+            None if data == "" else c_char_p(data.encode("UTF-8")))
+    if retval < 0:
+        errno = get_errno()
+        raise OSError(errno, strerror(errno))
+
+
+def main():
+    # type: () -> None
+    parser = ArgumentParser()
+    parser.add_argument("source", help="path of the device or bind source")
+    parser.add_argument("target", help="path of the new mount point")
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument("--bind", action="store_const", const="bind",
+                       help="create a new mount point of existing path")
+    group.add_argument("-t", "--type", help="filesystem type to mount")
+    opts = parser.parse_args()
+    if opts.bind is not None:
+        mount(opts.source, opts.target, b"", MountOpts.Bind)
```

Comment:
`mount(opts.source, opts.target, "", MountOpts.Bind)`? If you run this under Py3, `b"".encode('utf-8')` when calling mount will fail.

Reply:
What is the problem with b"".encode('utf-8')?

Reply:
`bytes` has no `encode()`, only `decode()`

Reply:
In python 2 bytes is just a string. It does have the encode method.
The b prefix makes it a plain string, not an unicode string. Note that this is the py2 version of the code.

Having said that I will look at unifying this even though I would much rather not have to use python 2 and all of its quirks at all.

Reply:
What I meant is that if you use the Py2 version of the script in Py3, `b""` is `bytes` which does not have `encode()`. I think it's the only line here that can cause compatibility problems.

IIRC `u""` is the unicode string in Py2, `b""` and `""` are the same (str). On the other hand, in Py3 `u""` is the same as `""` (str), but not `b""` (which is bytes).

Reply:
Yeah but this was for Python 2 specifically. The Python 3 version did not need that. 

In Python 3 u"" is invalid syntax. Look at what I did now, it works and is Mypy-clean on both versions.